### DISCUSSION TITLE
To what extent is writing TypeScript mandatory for users of Fresh?

### DISCUSSION THREAD

**Discussion by frou** [freshframework/fresh#299] [Q&A] [answered]
Is it considered a supported usage scenario for a user of Fresh to write all the code in `.js` & `.jsx` and never be obliged to write any TypeScript?

**Comment by LoganTann** [comment]
I think so but **as I never properly developed in this particular framework, I can't confirm**. (so please take a step back from my answer!!)

Deno has a flag `--no-check` to disable type checking. You can also add `// @ts-nocheck` at the beginning of every file you want to disable typescript.

Common uses cases are in my opinion to start the app faster in production when you know everything is type safe, or just because your codebase don't support typescript (due to a lack of architecture design from your side or just a poor typing compatibility with dependancies used). 

So my answer would be just to try it. However, chances are you're starting a new project so it's a very good opportunity to learn Typescript. Unless you have any particular reason to not use it, it worth the try. The learning curve is a bit difficult at first but I can confirm it changed in a positive way how I develop code with the js ecosystem.

**Comment by marvinhagemeister** [comment, marked as answer]
Yes, normal JavaScript is officially supported so you can use `.js` or `.jsx` files. The documentation is TS only for the moment and I hope we can add a way to switch between TS and JS in the code snippets there some time in the future. Either way, JS is supported.

**Reply by JoshuaWoodsdev** [reply]
How can i spin a project up without type?

**Reply by marvinhagemeister** [reply]
We don't have a non-TS template specifically, but you can bootstrap a Fresh project like you'd usually do, change the extensions and remove the types.
